### PR TITLE
Created gce_snapshot module

### PR DESCRIPTION
**Pull request by robwagner33**
##### ISSUE TYPE
 New Module Pull Request

##### COMPONENT NAME
modules/cloud/google/gce_snapshot

##### ANSIBLE VERSION
```
ansible 2.3.0 last updated 2017/01/09 13:28:32 (GMT -400)
```

##### SUMMARY
Added support for managing snapshots through the Google Cloud Platform. I've tried to match the style of the existing GCE modules. It is a very simple module that will allow you to create and destroy snapshots for an instance's storage volumes

### CODE REVIEW

**Comment by robwagner33** [comment]
Looks like Shippable is not able to connect to the IRC Server

Sending IRC notification  
No connection returned for server: [object Object]

**Comment by supertom** [comment]
@robwagner33 Thanks for the PR!

I see that disks is a list.  Should it be?  What's the expected behavior when multiple disks are specified?

**Comment by robwagner33** [comment]
Hey @supertom,

I think it should be a list, yes. The intent was to allow people to target a specific set of disks, so that if they had an instance with multiple volumes, they wouldn't necessarily have to snapshot all of them. For example, an instance has 4 volumes (disk0, disk1, disk2, disk3) but a user provides (disk1, disk2, disk3) so when the module runs, no snapshot will be created for disk0.

I'm not sure how often the option would need to be used, but it was just something that I thought of while developing.

Let me know if it would help to add an example, documentation or if you think the feature is unnecessary.

**Comment by supertom** [comment]
@robwagner33 Example would be great!

From what I see, if I were to specify multiple disks, wouldn't they all end up with the same snapshot name (which is invalid and would cause a ResourceExists exception I believe).  Is that true?

**Comment by robwagner33** [comment]
Ah, it looks like you are correct. I had made some local changes for a customization our company needed that auto-generates snapshot names for each disk and didn't re-test it thoroughly enough. 

I'll make the necessary changes and add an example with adding names for each disk.

**Comment by robwagner33** [comment]
I changed the logic to append the disk name when there is more than 1 disk attached. Now the disks list can be used if someone wants to only snapshot a particular subset of disks.

Please let me know if you think any additional changes are necessary.

File: lib/ansible/modules/cloud/google/gce_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+            elif snapshot and state == 'absent':
+                snapshot.destroy()
+                changed = True
+                msg = snapshot_name + " was deleted"
+
+            elif not snapshot and state == 'present':
+                volume_obj.snapshot(snapshot_name)
+                changed = True
+                msg = snapshot_name + " created"
```
Comment (supertom) [inline lib/ansible/modules/cloud/google/gce_snapshot.py]: As this is in a loop, the message will be overwritten on each iteration, with only the last snapshot created being reported.

File: lib/ansible/modules/cloud/google/gce_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    state: present
+    disks:
+      - disk0
+      - disk1
+    service_account_email: project_name@appspot.gserviceaccount.com
+    credentials_file: /path/to/credentials
+    project_id: project_name
+  delegate_to: localhost
+'''
+
```
Comment (supertom) [inline lib/ansible/modules/cloud/google/gce_snapshot.py:105]: Could you add a RETURN block?  You can find an example [here](https://github.com/ansible/ansible/blob/42e63d429c63e078e406d95edebd8f86cd44cfbd/lib/ansible/modules/cloud/google/gce_mig.py#L180)

**Review by supertom — CHANGES_REQUESTED** [review]
Code works, snapshots are created.

You do a great job of flagging what specifically happened when attempting to create a snapshot (already exists, created, etc).  I think we should turn those into lists (as opposed to just updating the message) that can be used downstream.

Think you could create lists for each of these?  I'm thinking snapshots_created, snapshots_deleted, snapshots_existing but I'm open.

**Comment by robwagner33** [comment]
Thanks for the feedback! I've made the changes you recommended.

**Comment by supertom** [comment]
@robwagner33 thanks for the PR!  LGTM!

**Review by ryansb — APPROVED** [review]

**Pull request merged**